### PR TITLE
noise-suppression-for-voice: init at 2020-10-12

### PR DESCRIPTION
**Pull request by henrikolsson**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
Did/does this actually work for you? I remember packaging this a while back and trying to use it with pulseaudio, because I was really impressed with RNNoise in mumble, but couldn't get it to work.

**Comment by henrikolsson** [comment]
> Did/does this actually work for you? I remember packaging this a while back and trying to use it with pulseaudio, because I was really impressed with RNNoise in mumble, but couldn't get it to work.

Yes, it works for me. Using it as suggested in the readme with pulseaudio loopback and monitor. My ~/.config/pulse/default.pa:
`.include /etc/pulse/default.pa

load-module module-null-sink sink_name=mic_denoised_out rate=48000
load-module module-ladspa-sink sink_name=mic_raw_in sink_master=mic_denoised_out label=noise_suppressor_mono plugin=/home/henrik/src/nix/noise-suppression-for-voice/result/lib/ladspa/librnnoise_ladspa.so control=50
load-module module-loopback source=alsa_input.usb-ZOOM_Corporation_H1n_000000000000-00.analog-stereo sink=mic_raw_in channels=1 source_dont_move=true sink_dont_move=true latency_msec=1

set-default-source mic_denoised_out.monitor
`

In what way didn't it work for you?

**Comment by srhb** [comment]
@ofborg eval

File: pkgs/development/libraries/noise-suppression-for-voice/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchgit, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "noise-supression-for-voice";
+  version = "2020-10-12";
```
Comment (srhb) [inline pkgs/development/libraries/noise-suppression-for-voice/default.nix:5]: According to: https://nixos.org/manual/nixpkgs/stable/#chap-reviewing-contributions :

>  If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23". 

So `pname` should have `-unstable` at the end.

File: pkgs/development/libraries/noise-suppression-for-voice/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchgit {
```
Comment (srhb) [inline pkgs/development/libraries/noise-suppression-for-voice/default.nix:7]: Prefer `fetchFromGitHub` here, for lightweight tarballs from github's API: https://nixos.org/manual/nixpkgs/stable/#sec-sources

File: pkgs/development/libraries/noise-suppression-for-voice/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    url = "https://github.com/werman/noise-suppression-for-voice";
+    sha256 = "11pwisbcks7g0mdgcrrv49v3ci1l6m26bbb7f67xz4pr1hai5dwc";
+    rev = "15bac8f34018184d7d4de1b3b2ba98c433705f6c";
+  };
+
+  buildInputs = [ cmake ];
```
Comment (srhb) [inline pkgs/development/libraries/noise-suppression-for-voice/default.nix:13]: https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages

`cmake` should be in `nativeBuildInputs` if it's a build-time only dependency. :slightly_smiling_face:

**Review by srhb — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've added a few requests for adjustments that I hope you're up for. :)

Also, let's see how ofborg feels about the `platforms` before merging with `all`

**Comment by srhb** [comment]
@ofborg build

**Comment by srhb** [comment]
@ofborg build noise-suppression-for-voice

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -14631,6 +14631,8 @@ in
 
   non = callPackage ../applications/audio/non { };
 
+  noise-supression-for-voice = callPackage ../development/libraries/noise-suppression-for-voice { };
```
Comment (srhb) [inline pkgs/top-level/all-packages.nix:14634]: Oh, typo here.

```suggestion
  noise-suppression-for-voice = callPackage ../development/libraries/noise-suppression-for-voice { };
```

**Comment by srhb** [comment]
@ofborg build noise-supression-for-voice

**Review by srhb — APPROVED** [review]
Thanks for the fixups!

**Pull request merged**